corehq/apps/sms/models.py

```diff
@@ -775,6 +777,67 @@ class Meta:
         app_label = "sms"
 
 
+class CommConnectCaseSQL(CommCareCaseSQL, CommCareMobileContactMixin):
+    class Meta:
+        app_label = "sms"
+        proxy = True
+
+    def get_phone_info(self):
+        PhoneInfo = namedtuple(
+            'PhoneInfo',
+            [
+                'requires_entry',
+                'phone_number',
+                'sms_backend_id',
+                'ivr_backend_id',
+            ]
+        )
+        contact_phone_number = self.get_case_property('contact_phone_number')
+        contact_phone_number = apply_leniency(contact_phone_number)
+        contact_phone_number_is_verified = self.get_case_property('contact_phone_number_is_verified')
+        contact_backend_id = self.get_case_property('contact_backend_id')
+        contact_ivr_backend_id = self.get_case_property('contact_ivr_backend_id')
+
+        requires_entry = (
+            contact_phone_number and
+            contact_phone_number != '0' and
+            not self.closed and
+            not self.is_deleted and
+            # For legacy reasons, any truthy value here suffices
+            contact_phone_number_is_verified
+        )
+
+        return PhoneInfo(
+            requires_entry,
+            contact_phone_number,
+            contact_backend_id,
+            contact_ivr_backend_id
+        )
+
+    def get_time_zone(self):
+        return self.get_case_property('time_zone')
+
+    def get_language_code(self):
+        return self.get_case_property('language_code')
+
+    def get_email(self):
+        return self.get_case_property('commcare_email_address')
+
+    @property
+    def raw_username(self):
+        return self.case_name
+
+    @classmethod
+    def wrap_as_commconnect_case(cls, case):
+        """
+        Takes a CommCareCaseSQL and returns it as a CommConnectCaseSQL.
+        """
+        # Having an extra lookup to do this isn't great.
+        # TODO: Try to replace all references to wrap_as_commconnect_case to
+        # avoid the extra lookup.
+        return CaseAccessorSQL.get_case(case.case_id, cls=cls)
```

Comment:
not sure i understand how to avoid the extra lookup since case is always going to be a CommCareCaseSQL, can you explain? or do you mean i'd just make a mixin for CommConnectCase and add it directly to the CommCareCaseSQL class?